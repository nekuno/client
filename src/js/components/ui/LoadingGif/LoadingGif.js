import React, { Component } from 'react';
import styles from './LoadingGif.scss';

export default class LoadingGif extends Component {
    render() {
        return (
            <img className={styles.loadinggif} src='/img/safari-pinned-tab.svg' alt='loading'/>
        );
    }
}