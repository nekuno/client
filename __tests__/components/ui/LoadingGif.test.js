import React from 'react';
import LoadingGif from '../../../src/js/components/ui/LoadingGif/LoadingGif.js';
import styles from '../../../src/js/components/ui/LoadingGif/LoadingGif.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test LoadingGif component', () => {
    const mockFn = jest.fn();
    const loadinggif = shallow(<LoadingGif onClickHandler={mockFn}>Lorem ipsum</LoadingGif>);

    it('should be defined', () => {
        expect(loadinggif).toBeDefined();
    });
    it('should render correctly', () => {
        expect(loadinggif).toMatchSnapshot();
    });
    it('LoadingGif click event', () => {
        loadinggif.find(`.${styles.loadinggif}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
