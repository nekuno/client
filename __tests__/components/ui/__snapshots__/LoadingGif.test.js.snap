// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoadingGif component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingGif
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </LoadingGif>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "loading",
      "className": "loadinggif",
      "src": "/img/safari-pinned-tab.svg",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "loading",
        "className": "loadinggif",
        "src": "/img/safari-pinned-tab.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
